iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractConnectable.java

```diff
@@ -178,7 +178,20 @@ protected void move(TerminalExt oldTerminal, TopologyPoint oldTopologyPoint, Str
         attachTerminal(oldTerminal, oldTopologyPoint, (VoltageLevelExt) bus.getVoltageLevel(), terminalExt);
     }
 
-    private void attachTerminal(TerminalExt oldTerminal, TopologyPoint oldTopologyPoint, VoltageLevelExt voltageLevel, TerminalExt terminalExt) {
+    void replaceTerminal(TerminalExt oldTerminal, TopologyPoint oldTopologyPoint, TerminalExt newTerminalExt, boolean notify) {
```

Comment:
This could probably be merged with attachTerminal, at the cost of adding one boolean and renaming it (I think it's more a replace than an attach). The referrer part in VoltageLevelImpl should also be included - that would fix the revealed bug when moving a connectable!

Reply:
Done